### DISCUSSION TITLE
Here is one nice dashboard

### DISCUSSION THREAD

**Discussion by stauffenberg2020** [jheredianet/Teslamate-CustomGrafanaDashboards#84] [Ideas]
Hey @jheredianet,

When you have some time, please check out https://github.com/tero-jarvinen/teslamate-speed-temperature

I hope someday all of your nice dashboards and this dashboard gets merged into the official teslamate dashboards

**Comment by jheredianet** [comment]
> Hey @jheredianet,
> 
> When you have some time, please check out https://github.com/tero-jarvinen/teslamate-speed-temperature
> 
> I hope someday all of your nice dashboards and this dashboard gets merged into the official teslamate dashboards

Hey @stauffenberg2020 it looks very nice. Thanks for sharing. About merging (about mine ones) of course I have it in mind. I'll do it step by step. I just began with the Battery Health dashboard and soon will merge other ones.

**Reply by stauffenberg2020** [reply]
When you have some time(Not urgent), can you help me a bit here, may be with some hint or directions on how to change the consumption units (wh/km) to be a standardized one as in your dashboards ...I tried to change the consumption units, But it looks like beyond my skill set. (I did try to update the 100/distance, But it did not work for every panel)

**Reply by jheredianet** [reply]
It dependes on the panel you are using. What panel do you want to change?

**Reply by stauffenberg2020** [reply]
I think all of the panels where the consumption is an output.

_Consumption at speed and temperature
Consumption at temperature
Short and long trip consumption at temperature
Consumption and temperature by date_

**Reply by jheredianet** [reply]
Ok, basically you need to have a Global variable (let's  say length_unit) on the dashboard to get the preference unit with:

```
SELECT preferred_range FROM settings LIMIT 1
```

Then on the dashboard based on that variable you can make a query like:
```
SELECT .....
 CASE 
    WHEN '$length_unit' = 'km' THEN .....
    WHEN '$length_unit' = 'mi' THEN .... 
  END AS END AS myfield_$length_unit
```

Then you may edit the panel and override the field by name  like "/.*_km/" and "/.*_mi/" so you can override any option like unit, display name, color, etc. 
I hope it helps.
